frame/staking/src/lib.rs

```diff
@@ -3227,12 +2392,17 @@ impl<T: Config> Module<T> {
 	/// Enact and process the election using the `ElectionProvider` type.
 	///
 	/// This will also process the election, as noted in [`process_election`].
-	fn enact_election(_current_era: EraIndex) -> Option<Vec<T::AccountId>> {
-		let _outcome = T::ElectionProvider::elect().map(|_| ());
-		log!(debug, "Experimental election provider outputted {:?}", _outcome);
-		// TWO_PHASE_NOTE: This code path shall not return anything for now. Later on, redirect the
-		// results to `process_election`.
-		None
+	fn enact_election(current_era: EraIndex) -> Option<Vec<T::AccountId>> {
+		T::ElectionProvider::elect()
+			.map_err(|e| log!(warn, "election provider failed due to {:?}", e))
+			.and_then(|(r, w)| {
```

Comment:
r, w is confusing with `read, write` here, I think `res, weight` are better names

Reply:
agreed.